include/orc/dwarf_structs.hpp

```diff
@@ -329,6 +316,7 @@ struct die {
     std::size_t _hash{0};
     std::size_t _fatal_attribute_hash{0};
     std::uint32_t _ofd_index{0}; // object file descriptor index
+    std::size_t _cu_die_address{0}; // address of associated compilation unit die entry
```

Comment:
This is added per-die, which was a cost I was waffling about, but the improvements are worth it.